pgx/src/datum/date.rs

```diff
@@ -41,7 +42,11 @@ impl FromDatum for Date {
         if is_null {
             None
         } else {
-            Some(datum.try_into().expect("Error converting date datum"))
+            if cfg!(feature = "pg10") {
+                Some(Date(datum.value() as i32))
```

Comment:
Oh deer.

Reply:
:upside_down_face: PostgreSQL has a... sometimes interesting way of handling Datums, yes.